plugins/crypto/mbedtls/certificategroup.c

```diff
@@ -132,6 +134,119 @@ MemoryCertStore_getRejectedList(UA_CertificateGroup *certGroup, UA_ByteString **
     return retval;
 }
 
+static UA_StatusCode
+checkCrlMatch(const UA_ByteString *certificate, const UA_ByteString *crl) {
+    UA_StatusCode retval = UA_STATUSCODE_GOOD;
+
+    mbedtls_x509_crt cert;
+    mbedtls_x509_crt_init(&cert);
+
+    mbedtls_x509_crl certRevocationList;
+    mbedtls_x509_crl_init(&certRevocationList);
+
+    UA_ByteString data1 = UA_mbedTLS_CopyDataFormatAware(certificate);
```

Comment:
I don't think you need to copy.
The `crt_parse` takes the length as an extra argument.
So it does not rely on a \0 at the end.

Reply:
The description of the function parameters includes the following.
"buflen – The size of buf, including the terminating NULL byte in case of PEM encoded data."

Reply:
I've added a helper function for mbedtls, similar to the one in OpenSSL, to load certificate files and CRLs. The function identifies the format of the input and only copies the byte string if it's in PEM format.